### PR TITLE
test: resolve all act errors

### LINKED ISSUE #334: Resolve all act errors
```plain
    console.error node_modules/react-dom/cjs/react-dom.development.js:88
      Warning: An update to EnvironmentProvider inside a test was not wrapped in act(...).
      
      When testing, code that causes React state updates should be wrapped into act(...):
      
      act(() => {
        /* fire events that update state */
      });
      /* assert on the output */
      
      This ensures that you're testing the behavior the user would see in the browser. Learn more at fb.me/react-wrap-tests-with-act
          in EnvironmentProvider (at testing-library.tsx:36)
          in I18nextProvider (at testing-library.tsx:35)
          in WithProviders (at testing-library.tsx:59)
          in RouterWrapper
```

Some tests are covered in these so they should be resolved to make tests more reliable and also reduce noise in logs in case some tests fail.

### PR DESCRIPTION
**Pull request by alesmit**
## Summary

Fixes #334

## Checklist

-   [x] Ready to be merged

### CODE REVIEW

File: src/app/components/Clipboard/ClipboardButton.test.tsx [hunk 1]
```diff
@@ -24,11 +24,9 @@ describe("ClipboardButton", () => {
 
 		expect(screen.queryByTestId("clipboard-button__checkmark")).not.toBeInTheDocument();
 
-		await act(async () => {
-			fireEvent.click(screen.getByTestId("clipboard-button__wrapper"));
-		});
+		fireEvent.click(screen.getByTestId("clipboard-button__wrapper"));
 
-		expect(screen.queryByTestId("clipboard-button__checkmark")).toBeInTheDocument();
+		expect(await screen.findByTestId("clipboard-button__checkmark")).toBeInTheDocument();
```
Comment (faustbrian) [inline src/app/components/Clipboard/ClipboardButton.test.tsx:29]:
```suggestion
		await expect(screen.findByTestId("clipboard-button__checkmark")).resolves.toBeInTheDocument();
```
Same for all of the others or it'll result in brittle expectations.

**Comment by faustbrian** [comment]
@alesmit conflicts

File: src/app/components/Clipboard/ClipboardButton.test.tsx (hunk 1 above)
Comment (alesmit) [reply src/app/components/Clipboard/ClipboardButton.test.tsx]: Fixed in https://github.com/PayvoHQ/wallet/pull/498/commits/92099cceb6d753e8ea4e1fa7d16756fb19fcdd75

**Comment by faustbrian** [comment]
The last commit can be reverted. They are being fixed and rules for it enabled in a separate PR to keep this PR strictly to a single issue and rule.

**Comment by alesmit** [comment]
@faustbrian tests in the setting domain are flaky and fail depending on which test block runs first. The issue is not related to this PR scope, so I think we can leave it and I will fix it with another PR. Then there's missing coverage for one line in the transaction domain which needs further investigation, and yet would be good to use a different PR for it.

**Comment by faustbrian** [comment]
There's some missing coverage in several domains as a result of changes here so that should be resolved here before we merge this.

**Review by faustbrian — APPROVED** [review]

**Pull request merged**